FirebaseAuthExchange/Sources/Models/ExchangeServiceModels.swift

```diff
@@ -60,3 +60,66 @@ struct ExchangeOIDCTokenRequest: Encodable {
     self.authCodeCredentials = authCodeCredentials
   }
 }
+
+// MARK: - Response Models
+
+/// Response that encapsulates an Auth Exchange access token.
+///
+/// This is the return value for an `ExchangeInstallationAuthTokenRequest` or
+/// `ExchangeCustomTokenRequest`.
+struct ExchangeTokenResponse: Decodable, Equatable {
+  /// A container for a Firebase  access token and the duration of time until it expires.
+  struct AuthExchangeToken: Decodable, Equatable {
+    /// A signed [JWT](https://tools.ietf.org/html/rfc7519) containing claims that identify a user.
+    let accessToken: String
+
+    /// The duration of time until the `accessToken` expires, approximately relative to the time
+    /// this response was received.
+    let timeToLive: ProtobufDuration
+
+    enum CodingKeys: String, CodingKey {
+      case accessToken
+      case timeToLive = "ttl"
+    }
+  }
+
+  /// An Auth Exchange access token that can be used to access Firebase services.
+  let token: AuthExchangeToken
+}
+
+/// Model of a `google.protobuf.Duration`, which represents a time span.
+///
+/// A Protocol Buffer
+/// [`Duration`](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#duration)
+/// represents a signed, fixed-length span of time represented as a count of seconds and fractions
+/// of seconds at nanosecond resolution. It is independent of any calendar and concepts like "day"
+/// or "month".
+struct ProtobufDuration: Equatable {
+  private static let nanosecondsPerSecond = 1e9 // 1 x 10⁹ nanoseconds / second
+
+  /// The number of seconds in the time span.
+  let seconds: Int64
+
+  /// The fraction of a second, at nanosecond resolution, in addition to `seconds` in the time span.
+  let nanoseconds: Int32
+
+  /// Floating-point representation of the time span in seconds.
+  var duration: TimeInterval {
+    return TimeInterval(seconds) + Double(nanoseconds) / ProtobufDuration.nanosecondsPerSecond
+  }
+
+  /// Creates a new instance from a JSON string representation of a `Duration`.
+  ///
+  /// - parameter json: a string with the format "`{seconds}.{nanoseconds}s`", with nanoseconds
+  ///   expressed as fractional seconds.
+  init(json: String) throws {
+    (seconds, nanoseconds) = try parseDuration(text: json)
+  }
+}
```

Comment:
This might be a basic question, but would you be able to just use a class like [`UnitDuration`](https://developer.apple.com/documentation/foundation/nsunitduration?language=objc) (which seems to support nanoseconds) instead of making a new struct here?

Reply:
To be honest I hadn't ever run across the [Units and Measurements](https://developer.apple.com/documentation/foundation/units_and_measurement) APIs in `Foundation` so thanks for pointing this out. It's nice that it handles the units when doing operations like addition, e.g., `Measurement(value: 1.0, unit: UnitDuration.hours) + Measurement(value: 5, unit: UnitDuration.minutes)`. Unfortunately, [`UnitSeconds.nanoseconds`](https://developer.apple.com/documentation/foundation/unitduration/3172540-nanoseconds) was added in iOS 13.

We may want to revisit this decision in the future when we're targeting iOS 13+ but for now I think a custom struct is simpler. In an ideal world we'd use a [`Duration`](https://developer.apple.com/documentation/swift/duration) since it's a value (struct) type and supports nanoseconds but it was only added in iOS 16 :cry:.